frontend/src/metabase/actions/components/ActionViz/Action.unit.spec.tsx

```diff
@@ -0,0 +1,242 @@
+import React from "react";
+import _ from "underscore";
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+import { waitFor } from "@testing-library/react";
+
+import { render, screen } from "__support__/ui";
+
+import {
+  createMockActionParameter,
+  createMockQueryAction,
+  createMockImplicitQueryAction,
+} from "metabase-types/api/mocks";
+
+import { ActionComponent, ActionProps } from "./Action";
+
+const defaultProps = {
+  dashcard: {
+    id: 456,
+    card_id: 777, // action model id
+    action: createMockQueryAction({
+      name: "My Awesome Action",
+      parameters: [
+        createMockActionParameter({
+          id: "1",
+          name: "Parameter 1",
+          type: "type/Text",
+          target: ["variable", ["template-tag", "1"]],
+        }),
+        createMockActionParameter({
+          id: "2",
+          name: "Parameter 2",
+          type: "type/Text",
+          target: ["variable", ["template-tag", "2"]],
+        }),
+      ],
+    }),
+    parameter_mappings: [
+      {
+        parameter_id: "dash-param-1",
+        card_id: 1,
+        target: ["variable", ["template-tag", "1"]],
+      },
+      {
+        parameter_id: "dash-param-2",
+        card_id: 1,
+        target: ["variable", ["template-tag", "2"]],
+      },
+    ],
+  },
+  dashboard: { id: 123 },
```

Comment:
Can we use `createMockDashboard` utility to make it closer to a real environment?